includes/class-newspack-popups-inserter.php

```diff
@@ -968,5 +1035,58 @@ public static function should_display( $popup, $check_if_is_post = false ) {
 
 		return $is_post_context_matching;
 	}
+
+	/**
+	 * Add an admin bar button for logged-in admins and editors to toggle Campaigns visibility.
+	 */
+	public static function add_preview_toggle() {
+		if ( ! self::should_show_admin_ui() ) {
+			return;
+		}
+
+		global $wp_admin_bar;
+		$wp_admin_bar->add_menu(
+			[
+				'parent' => false,
+				'id'     => 'campaigns_preview_toggle',
+				'title'  => __( 'Prompts Visible', 'newspack-popups' ),
+				'href'   => '#',
+				'meta'   => [
+					'class' => 'newspack-campaigns-preview-toggle',
+				],
+			]
+		);
+	}
+
+	/**
+	 * Add empty DOM elements to avoid admin UI class names from being stripped by AMP.
+	 */
+	public static function insert_admin_ui_elements() {
+		if ( ! Newspack_Popups_Segmentation::is_admin_user() || ! self::is_amp_plus() ) {
+			return;
+		}
+		?>
+			<div class="newspack-popups-hide-prompts"></div>
+		<?php
+	}
+
+	/**
+	 * Allow admin UI scripts to be loaded in AMP Plus mode.
+	 *
+	 * @param bool|null $is_sanitized If null, the error will be handled. If false, rejected.
+	 * @param object    $error        The AMP sanitisation error.
+	 *
+	 * @return bool Whether the error should be rejected.
+	 */
+	public static function filter_scripts_for_amp_plus( $is_sanitized, $error ) {
+		if (
+			method_exists( '\Newspack\AMP_Enhancements', 'is_script_id_matching_strings' ) &&
+			\Newspack\AMP_Enhancements::is_script_id_matching_strings( [ self::ADMIN_SCRIPT_HANDLE ], $error )
```

Comment:
The localisation script has to be filtered for, too. Best to filter by script containing the `newspack_popups_admin` string (`is_script_body_matching_strings` method), which both scripts will contain.

Reply:
Like this? e708859

Reply:
Precisely!